src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionImportFunction.cs

```diff
@@ -35,10 +30,7 @@ public class CompleteNextDataSetVersionImportFunction(
     {
         return await requestValidator
             .Validate(request, cancellationToken)
-            .OnSuccess(_ => GetNextDataSetVersionInMappingStatus(request, cancellationToken))
-            .OnSuccessDo(_ => GetCompletedDataSetVersionMapping(request, cancellationToken))
-            .OnSuccessCombineWith(nextDataSetVersion =>
-                GetImportInManualMappingStage(request, nextDataSetVersion))
+            .OnSuccess(_ => mappingService.GetManualMappingVersionAndImport(request, cancellationToken))
```

Comment:
I think I'd have moved the logic in the `.OnSuccess` below this into this `GetManualMappingVersionAndImport` method too. So that the service returns the view model. And I'd have named the method `CompleteNextDataSetVersionImport` or something instead.

Reply:
So all the function ends up doing is  ==> receiving the request ==> delegating the business logic to the service layer, which returns a viewmodel ==> then the function handles the response

Reply:
A couple of examples are CreateDataSetFunction and CreateNextDataSetVersionMappingsFunction

Reply:
I think `CreateDataSetFunction` and `CreateNextDataSetVersionMappingsFunction` could still be changed to return the view model in the service layer though couldn't they (not in this PR)?

Reply:
And what to do with the code which kicks off the main orchestration following the completion of the http trigger?  Would you suggest that goes in the service layer also?  Feels to me that you'd wanna retain that in the function code personally.

Reply:
I think I'd still do that part in the function code! I'd agree it feels like it belongs in that layer